compiler/src/main/java/com/squareup/anvil/compiler/Utils.kt

```diff
@@ -258,3 +258,21 @@ internal fun AnnotationDescriptor.requireClass(): ClassDescriptor {
     message = "Couldn't find the annotation class for $fqName",
   )
 }
+
+/**
+ * This function should only be used for package names. If the FqName is the root (no package at
+ * all), then this function returns an empty string whereas `toString()` would return "<root>". For
+ * a more convenient string concatenation the returned result can be prefixed and suffixed with an
+ * additional dot. The root package never will use a prefix or suffix.
+ */
+internal fun FqName.safePackageString(
+  dotPrefix: Boolean = false,
+  dotSuffix: Boolean = true
+): String =
+  if (isRoot) {
+    ""
+  } else {
+    val prefix = if (dotPrefix) "." else ""
+    val suffix = if (dotSuffix) "." else ""
+    "$prefix$this$suffix"
```

Comment:
This looks awkward, is there a more idiomatic way that's also readable?

Reply:
Hard to say without having some sample input to look at. I assume some instances return something like .com.squareup.foo and some return com.squareup.bar.

Since your usages of safePackageString always include a suffix, I'd remove that as a parameterized option. Also, you can detect if the string is already prefixed with a dot and add it if not, since that appears to be the intended purpose. That'll clean up your API since you'll take care of it via code instead of having the dev determine proper usage.

`
internal fun FqName.safePackageString(): String =
      if(isRoot) {
          ""
      } else {
         ".$this.removeSurrounding(".")."
       }     
`

Reply:
The string is never prefixed with a dot. The two parameters make string templates easier to use. Imagine `prefix.package.${realPackage}.suffix`. Notice that for the root package this would result in `prefix.package..suffix`, which is annoying to handle. Therefore, this method accepts two parameters whether there is prefix or suffix for the template.